WPFHospitalEditor/SearchResultDialog.xaml.cs

```diff
@@ -138,6 +150,57 @@ private void addAdvancedSearchButton()
             Grid.SetColumn(advancedSearch, 4);
 
             DynamicGrid.Children.Add(advancedSearch);
+            advancedSearch.Click += (s, e) =>
+            {
+                
+                if (row.ContainsKey(Grid.GetRow(advancedSearch)))
+                {
+                    MapObject chosenMapObject = row[Grid.GetRow(advancedSearch)];
+                    selectedObjectId = row[Grid.GetRow(advancedSearch)].Id;
+                    mapObjectController.update(chosenMapObject);
+
+                    if (chosenMapObject.Description.Equals(""))
+                    {
+                        CanvasService.addObjectToCanvas(mapObjectController.getOutterMapObjects(), HospitalMap.canvasHospitalMap);
+                        this.Close();
+                    }
+                    else
+                    {
+                        String building = getBuildingAndFloor(chosenMapObject).Item1;
+                        String floor = getBuildingAndFloor(chosenMapObject).Item2;
+                        List<MapObject> chosenBuilding = findBuilding(building);
+                        displayBuildingAndFloorBasedOnSelectedObject(chosenBuilding, int.Parse(floor), role, int.Parse(building));
+                        
+                        hospitalMap.Hide();
+                        this.Close();
+                    }
+                }
+            };
+        }
+        
+        public void displayBuildingAndFloorBasedOnSelectedObject(List<MapObject> chosenBuilding,int  floor, Role role, int building)
+        {
+            Building buildingFromSearch = new Building(chosenBuilding, floor, role);
+            Building.canvasBuilding.Children.Clear();
+            CanvasService.addObjectToCanvas(getObjects(building.ToString(), floor.ToString()), Building.canvasBuilding);
+            buildingFromSearch.Owner = hospitalMap;
+            buildingFromSearch.Show();
+        }
+        public List<MapObject> findBuilding(String building)
+        {
+            string[] split1;
+            string[] split2;
+            List<MapObject> buildingObjects = new List<MapObject>();
+            foreach (MapObject maoObjectIterate in mapObjectController.getAllMapObjects())
+            {
+                split1 = maoObjectIterate.Description.Split("&");
+                split2 = split1[0].Split("-");
```

Comment:
Try not to repeat the code. You already have a function that separates this data (getBuildingAndFloor ), use it. The problem may arise later, for example in the case of the way the objects are placed in the file. In that case, you will have to change the code in many more places than just one.

Reply:
Fixed.